chapter2.adoc

```diff
@@ -56,32 +56,32 @@ Debug operations in all modes are prohibited for hart i when the logical-OR of `
 [options="header"]
 .The debuggable privilege levels per debug controls 
 |============================================
-| mdbgen | sdedbgalw | Debuggable privilege levels 
+| mdbgen[i] | sdedbgalw | Debuggable privilege levels 
 | 1      | x      | All                      
 | 0      | 1      | All except M             
 | 0      | 0      | n/a                      
 |============================================
 
 
 === Trace
-The extension requires that trace availability from each hart is constrained by default. When Zedsec is supported, the optional sideband signal to trace encoder, sec_check cite:[etrace], must be implemented for each hart, and this signal must be reset to 1. The sec_check signal is only cleared when trace is permitted by machine mode trace control or submachine mode trace control.
+The extension requires that trace availability from each hart is constrained by default. When Zedsec is supported, the optional sideband signal to trace encoder, sec_check[i] cite:[etrace], must be implemented for each hart i, and this signal must be reset to 1. The sec_check[i] signal is only cleared when trace is permitted by machine mode trace control or submachine mode trace control.
 
 ==== Machine Mode Trace Control 
-For each hart i, an input port, mtrcen[i], controls machine mode trace availability. Setting mtrcen[i] to 1 enables machine mode and submachine mode trace by clearing the sec_check signal to 0 across all privilege levels. Conversely, if mtrcen[i] is set to 0, the sec_check signal cannot be cleared when the hart operates in machine mode.
+For each hart i, an input port, mtrcen[i], controls machine mode trace availability. Setting mtrcen[i] to 1 enables machine mode and submachine mode trace by clearing the sec_check[i] signal to 0 across all privilege levels. Conversely, if mtrcen[i] is set to 0, the sec_check[i] signal cannot be cleared when the hart operates in machine mode.
```

Comment:
@AoteJin, I understand mtrcen and mdbgen signals are defined as per-hart signal to target heterogonous Multi Core Systems . Can we please add clarification for homogenous cores in the text? Something like this one: mtrcen/mdbgen input ports can be shared for homogenous harts as global signal instead of per-hart input

Reply:
Sure, I think I could add extra note in next commit. Something like: for homogenous system, the implementer could bundle all mdbgen[I] to a unified port, similarly to mtrce[I]. How do you like it?

Reply:
Great, it makes sense to me. It will clarify for homogenous multi-core systems.